### PR TITLE
feat(backend/shared): add action buttons to tasks bb-362

### PR DESCRIPTION
**Pull request by marharita08**
## Summary
Added backend part for 'adding action buttons for task' #362.
### backend
PATCH tasks/update - to update status of task 
GET tasks/past - to get tasks with status 'Completed' or 'Skipped' by id of current user

## Added
- typings and validation schema to update task endpoint
- preHandler to check that user updates own task

### CODE REVIEW

File: apps/backend/src/modules/tasks/libs/hooks/check-access-to-task.hook.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ErrorMessage } from "~/libs/enums/enums.js";
+import { type APIPreHandler } from "~/libs/modules/controller/libs/types/types.js";
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/libs/hooks/check-access-to-task.hook.ts:2]:
```suggestion
import { type APIPreHandler } from "~/libs/modules/controller/controller.js";
```

File: apps/backend/src/modules/tasks/libs/hooks/check-access-to-task.hook.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		const taskId = (params as TaskUpdateParametersDto).id;
+
+		const task = await taskService.find(taskId);
+
+		if ((user as UserDto).id !== task?.userId) {
+			throw new TaskError({
+				message: ErrorMessage.FORBIDDEN,
+				status: HTTPCode.FORBIDDEN,
+			});
+		}
```
Comment (fshabanov) [inline apps/backend/src/modules/tasks/libs/hooks/check-access-to-task.hook.ts:24]: Shouldn't you call `done()` at some point?

**Review by fshabanov — CHANGES_REQUESTED** [review]
Add screenshots, please

File: apps/backend/src/modules/tasks/libs/hooks/check-access-to-task.hook.ts (hunk 2 above)
Comment (marharita08) [reply apps/backend/src/modules/tasks/libs/hooks/check-access-to-task.hook.ts:24]: I got a message that `done()` shouldn't be called in async preHandler

**Review by fshabanov — APPROVED** [review]

**Comment by fshabanov** [comment]
Merge the main branch and we can merge the Backend part

**Pull request merged**